source/arm7/libc/crt0_device_list.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Zlib
+//
+// Copyright (C) 2025 Edoardo Lolletti (edo9300)
+
+#include <stddef.h>
+#include <string.h>
+
+#include <nds/system.h>
+#include <nds/device_list.h>
+
+static size_t getDeviceNameLenFromAppName(const char (appname)[0x40])
+{
+	for(int i = 0; i < 0x40; ++i)
+	{
+		if(appname[i] == ':')
+		{
+			return i;
+		}
+	}
+	return 0;
+}
+
+struct device_list_argv {
```

Comment:
`TWL_BSS` or turn this into a type and `malloc()` it

Reply:
This struct is not instantiated with a variable, it's only used as type to access the address __system_argv is at

Reply:
Oh, sorry!